_pylibczi/Reader.cpp

```diff
@@ -238,13 +232,28 @@ namespace pylibczi {
               c = vi[i].first;
               condensed.insert(vi[i].second);
           }
-          charSizes.emplace_back(c, condensed.size());
+          // added special case for images with multiple channels present
+          if (c=='C' && condensed.size()==1 && image_shape.size()==3)
+              charSizes.emplace_back(c, image_shape.end()[-3]); // channels
+          else
+              charSizes.emplace_back(c, condensed.size());
           condensed.clear();
       }
-      auto heightByWidth = images_.front()->shape(); // assumption: images are the same shape, if not 🙃
-      charSizes.emplace_back('Y', heightByWidth[0]); // H
-      charSizes.emplace_back('X', heightByWidth[1]); // W
+      charSizes.emplace_back('Y', image_shape.end()[-2]); // H
+      charSizes.emplace_back('X', image_shape.end()[-1]); // W
```

Comment:
This ties into support for "inline" BGR encoded czifiles; see also my comments regarding the ``flatten`` flag. In my opinion, the negative indices are no more dangerous than hard coded positive indices, and here they are useful because if it is a BGR file, the order is: (1) color channels (2) height and (3) width, but if it's grayscale, the order is: (1) height and (2) width. So counted from the end, width is always the last index and height is always the second-to-last, but counting from the beginning it depends on whether it is a color encoded image or not. But again, if you can get my sample non-mosaic BGR sample file to load properly with a cleaner fix, I'd be happy with that.